.github/actions/find-intermediates-base/README.md

```diff
@@ -0,0 +1,40 @@
+# Find Intermediates Base Action
+
+This GitHub Action finds the most recent commit (from the last 5) with cached build artifacts for a specified Nix flake `packages` output.
+
+## Dependencies
+
+- **Nix**: Required for checking if the binary cache has intermediates cached.
+- **Git**: Required for `git log` to retrieve commit history.
+- **Bash**: The Action uses Bash scripting, so a Unix-like environment is required.
+
+
+### Example Workflow
+
+```yaml
+name: CI with Cached Intermediates
+on:
+  pull_request:
+jobs:
+  find-intermediates:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v1.0.0
+      - name: Find Build Intermediates Base
+        if: github.event_name == 'pull_request'
+        id: find_build_intermediates
+        uses: ./.github/actions/find-intermediates-base
+        with:
+          cache_url: "https://cache.nixos.org"
+      - name: Build Incrementally
+        run: |
+          if [ -n "${{ steps.find_build_intermediates.outputs.commit_hash }}" ]; then
+            echo "Building incrementally from commit ${{ steps.find_build_intermediates.outputs.commit_hash }}"
+            # Add your incremental build command here
```

Comment:
Since this is example code after all, could you actually use an example build command here? 

```
nix build --override-input ...
```

(Same below)

Reply:
https://github.com/juspay/haskell-incremental-build-template/pull/5/commits/e6ab0971b14e21d14a4fc6d0eb1490f46a78aa5c